docs/examples/models_generics_inheritance_extend.py

```diff
@@ -0,0 +1,17 @@
+from typing import TypeVar, Generic
+from pydantic.generics import GenericModel
+
+TypeX = TypeVar('TypeX')
+TypeY = TypeVar('TypeY')
+
+
+class BaseClass(GenericModel, Generic[TypeX]):
+    x: TypeX
+
+
+class ChildClass(BaseClass[int], Generic[TypeY]):
+    y: TypeY
```

Comment:
See the comment below.  Would this also work?
I would think that this would be the "new feature" that requires documentation.  
While the above should just work anyway (and thanks to your fix it now does ;))

```python
TypeX = TypeVar('TypeX')
TypeY = TypeVar('TypeY')
TypeZ = TypeVar('TypeZ')



class BaseClass(GenericModel, Generic[TypeX, TypeX]):
    x: TypeX
    y: TypeY


class ChildClass(BaseClass[int, TypeY], Generic[TypeY, TypeZ]):
    z: TypeZ

# x is type int, y and z are type str
print(ChildClass[str, str](x=1, y='y', z='z'))

```

Reply:
Yeah, that should work once #1989 is merged. Unfortunately, the update to the documentation doesn't really fit in either of our PRs, since this example won't work without both.

Should the doc update happen in a separate PR after both are merged, or should we just include it in one of these PRs? I favor updating the documentation in this PR, so it doesn't get forgotten. The documentation should outline how pydantic is _supposed_ to work, right?

Also, I like your example better than mine. If we decide to keep the documentation update here, then I'll update my example to match.

Reply:
@choogeboom @daviskirk Great job guys on all the generic part 👍  I agree with you I feel like the doc should be updated in one of your PRs, probably this one as I would first merge #1989 too.